### PR TITLE
Fix "container is falsey" error when using code coverage

### PR DESCRIPTION
**Pull request by mdeanjones**
#112

### CODE REVIEW

**Comment by mdeanjones** [comment]
Node 6 does not like object spreads.

File: fixtures/istanbul-should-cover.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable */
```
Comment (rwjblue) [inline fixtures/istanbul-should-cover.js:1]: Let's move the fixture into the test file (and narrow as much of the contents as possible to still reproduce the errors)

**Review by rwjblue — COMMENTED** [review]
Thanks for working on this!

**Comment by mdeanjones** [comment]
Ya, I had some misconceptions about how the Istanbul plugin was getting things done.

File: __tests__/index-test.js [hunk 2]
```diff
@@ -422,3 +422,25 @@ describe('when used with typescript', () => {
     expect(actual).toEqual(`Ember.addObserver();`);
   });
 });
+
+describe('when used with babel-plugin-istanbul', () => {
+  // babel-plugin-istanbul won't run on <= Node 6
+  const majorVersion = parseInt(process.version.match(/^v(\d+)\./)[1], 10);
+  const runOrSkip = majorVersion > 6 ? it : it.skip;
```
Comment (rwjblue) [inline __tests__/index-test.js:429]: Nice one, thank you!

File: __tests__/index-test.js [hunk 3]
```diff
@@ -422,3 +422,25 @@ describe('when used with typescript', () => {
...
+
+      export default class TestObject extends EmberObject.extend(Evented) {};
+    `;
+
+    expect(() => {
+      babel7.transformSync(source, {
+        filename: 'istanbul-should-cover.js',
+        plugins: [require('babel-plugin-istanbul'), Plugin],
+      });
+    }).toThrow(/Container is falsy/i);
```
Comment (rwjblue) [inline __tests__/index-test.js:444]: Can you tweak this to something like:

```suggestion
    expect(() => {
      babel7.transformSync(source, {
        filename: 'istanbul-should-cover.js',
        plugins: [require('babel-plugin-istanbul'), Plugin],
      });
    }).toMatchInlineSnapshot();
```

Basically, we **don't** want it to throw...

Comment (mdeanjones) [reply __tests__/index-test.js:444]: To make sure I'm tracking, the idea is to provide feedback that's a bit more useful than "it did/not throw"?

Comment (rwjblue) [reply __tests__/index-test.js:444]: Right, exactly. Basically `expect().toThrow()` is not what we want here, we want to say it **does not** throw (IOW, this PR should be failing not passing). Using `.toMatchInlineSnapshot` just lets us confirm that it emits _something_ that we can visually verify "looks correct".

**Comment by mdeanjones** [comment]
Poking some more, it looks like [this is the root cause](https://github.com/babel/babel/issues/7596#issuecomment-388075523).

**Comment by mdeanjones** [comment]
Standing on bigger shoulders than my own, this _seems_ like a commonly accepted solution - https://github.com/istanbuljs/babel-plugin-istanbul/pull/192

Don't know if it's the most efficient place to put it though.

File: src/index.js [hunk 4]
```diff
@@ -165,6 +165,8 @@ module.exports = function (babel) {
                 ])
               );
             } else {
+              path.scope.crawl();
```
Comment (rwjblue) [inline src/index.js:168]: I worry that doing this for **every** import declaration replacement is going to be quite costly. Now that you fully understand what is wrong, I wonder if we can avoid the re-crawl by detecting the specific failure scenario and handling it. 🤔 

Thoughts @mdeanjones?

Comment (mdeanjones) [reply src/index.js]: Makes sense. I opted for a "_try, and try again_" approach. It gets the job done, but does feel a bit... hacky.

**Comment by rwjblue** [comment]
Thank you @mdeanjones!

**Pull request merged**